settings.py

```diff
@@ -44,27 +58,44 @@
 LOGGING['root']['level'] = env('DJANGO_LOG_LEVEL', 'INFO')
 
 AWS_STORAGE_BUCKET_NAME = env('AWS_STORAGE_BUCKET_NAME', '')
+AWS_BUCKET_DOMAIN = env('AWS_BUCKET_DOMAIN', AWS_STORAGE_BUCKET_NAME + '.s3.amazonaws.com')
 CDN_DOMAIN_NAME = env('CDN_DOMAIN_NAME', '')
+AWS_ACCESS_KEY_ID = env('AWS_ACCESS_KEY_ID', '')
+AWS_SECRET_ACCESS_KEY = env('AWS_SECRET_ACCESS_KEY', '')
+AWS_S3_REGION_NAME = env('AWS_S3_REGION_NAME', '')
+AWS_DEFAULT_ACL = env('AWS_DEFAULT_ACL', '')
+AWS_LOCATION = env('AWS_LOCATION', '')
+AWS_STATIC = env('AWS_STATIC', False)
+AWS_MEDIA = env('AWS_MEDIA', False)
 
 if AWS_STORAGE_BUCKET_NAME:
     # Tell django-storages that when coming up with the URL for an item in S3 storage, keep
     # it simple - just use this domain plus the path. (If this isn't set, things get complicated).
     # This controls how the `static` template tag from `staticfiles` gets expanded, if you're using it.
     # We also use it in the next setting.
     if CDN_DOMAIN_NAME:
-        AWS_S3_CUSTOM_DOMAIN = CDN_DOMAIN_NAME
+        AWS_S3_DOMAIN = CDN_DOMAIN_NAME
```

Comment:
Why change this? `AWS_S3_CUSTOM_DOMAIN` is a variable known to django-storages:

http://django-storages.readthedocs.io/en/latest/backends/amazon-S3.html#cloudfront

Reply:
@tobiasmcnulty The custom domain gets used here:
https://github.com/jschneier/django-storages/blob/master/storages/backends/s3boto3.py#L562

You will see it then returns before generating a presigned URL which we needed:
https://github.com/jschneier/django-storages/blob/master/storages/backends/s3boto3.py#L571

Reply:
@erikh360 Yes...and my point is that that setting goes by the name `AWS_S3_CUSTOM_DOMAIN`, not `AWS_S3_DOMAIN`, in django-storages: https://github.com/jschneier/django-storages/blob/master/storages/backends/s3boto3.py#L196

Reply:
@tobiasmcnulty We needed a presigned URL to access S3. 

If we set the `AWS_S3_CUSTOM_DOMAIN` setting it is assigned to `custom_domain` on L196, this causes the `url()` function to return a URL on L562 that is not presigned.

We want it to get to L571 to generate the presigned URL and this will only happen if `AWS_S3_CUSTOM_DOMAIN` is not set.

Reply:
Ah, perhaps add a comment to that effect then. Took me awhile to figure out that you were using a different variable name intentionally. How about something like:

    # If we call this setting `AWS_S3_CUSTOM_DOMAIN`, that breaks presigned URLs in
    # django-storages. Use our own setting for the domain instead, which is unknown to
    # django-storages.